hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/Sinks.java

```diff
@@ -675,7 +682,57 @@ private Sinks() {
      * @param <T> type of the items the sink accepts
      */
     @Nonnull
-    public static <T> JmsSinkBuilder<T> jmsTopicBuilder(DistributedSupplier<ConnectionFactory> factorySupplier) {
-        return new  JmsSinkBuilder<>(factorySupplier, true);
+    public static <T> JmsSinkBuilder<T> jmsTopicBuilder(@Nonnull DistributedSupplier<ConnectionFactory> factorySupplier) {
+        return new JmsSinkBuilder<>(factorySupplier, true);
+    }
+
+    /**
+     * Returns a sink that connects to the specified database using the given
+     * {@code connectionSupplier} and writes to it using the given {@code
+     * statementFn} and {@code updateFn}. After a batch of items is written,
+     * {@code flushFn} is called to flush the changes.
+     * <p>
+     * No state is saved to snapshot for this sink. After the job is restarted,
+     * the items will likely be duplicated, providing an <i>at-least-once</i>
+     * guarantee.
+     * <p>
+     * The default local parallelism for this sink is 1.
+     * @param connectionSupplier the supplier of database connection
+     * @param statementFn the function to prepare the statement
+     * @param updateFn the function to set the parameters of the statement for
+     *                 each item received and execute the update query. Instead
+     *                 of updating for each item, it can be used to add to batch
+     *                 via {@link PreparedStatement#addBatch()}
+     * @param flushFn the function to flush the changes. Gets {@code connection}
+     *                and {@code statement} as arguments. It can be used to
+     *                commit the connection if it is not {@code autoCommit} mode
+     *                or execute the batch
+     * @param <T> type of the items the sink accepts
+     */
+    @Nonnull
+    public static <T> Sink<T> jdbc(
+            @Nonnull DistributedSupplier<Connection> connectionSupplier,
+            @Nonnull DistributedFunction<Connection, Statement> statementFn,
+            @Nonnull DistributedBiConsumer<Statement, T> updateFn,
+            @Nonnull DistributedBiConsumer<Connection, Statement> flushFn
+    ) {
+        return Sinks.fromProcessor("jdbcSink",
+                SinkProcessors.writeJdbcP(connectionSupplier, statementFn, updateFn, flushFn));
+    }
+
+    /**
+     * Convenience for {@link Sinks#jdbc(DistributedSupplier,
+     * DistributedFunction, DistributedBiConsumer, DistributedBiConsumer)}.
+     */
+    @Nonnull
+    public static <T> Sink<T> jdbc(@Nonnull String connectionUrl,
```

Comment:
We also need driver class here and do `Class.forName(driverClass)` before calling `DriverManager.getConnection`. JDBC drivers register themselves in a static initializer in the driver class, if we don't do this, the connection will fail. The test doesn't fail because the `DeleteDbFiles.execute` likely loads the driver.

Reply:
`In previous versions of JDBC, to obtain a connection, you first had to initialize your JDBC driver by calling the method Class.forName. This methods required an object of type java.sql.Driver. Each JDBC driver contains one or more classes that implements the interface java.sql.Driver.
...
Any JDBC 4.0 drivers that are found in your class path are automatically loaded. (However, you must manually load any drivers prior to JDBC 4.0 with the method Class.forName.)`
JDBC 4.0 is introduced with Java6

Reply:
Ok, didn't know that. Maybe you should test whether it works with the dynamic class loading when the driver is submitted in job resources.

Reply:
I'll test it